pkg/autodiscovery/providers/cloudfoundry.go

```diff
@@ -0,0 +1,216 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+//
+// +build clusterchecks
+
+package providers
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/providers/names"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/cloudfoundry"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/bhmj/jsonslice"
+)
+
+// CloudFoundryConfigProvider implements the Config Provider interface, it should
+// be called periodically and returns templates from Cloud Foundry BBS for AutoConf.
+type CloudFoundryConfigProvider struct {
+	bbsCache      cloudfoundry.BBSCacheI
+	lastCollected time.Time
+}
+
+// NewCloudFoundryConfigProvider instantiates a new CloudFoundryConfigProvider from given config
+func NewCloudFoundryConfigProvider(conf config.ConfigurationProviders) (ConfigProvider, error) {
+	cfp := CloudFoundryConfigProvider{
+		lastCollected: time.Now(),
+	}
+	var err error
+
+	cfp.bbsCache, err = cloudfoundry.GetGlobalBBSCache()
+	if err != nil {
+		return nil, err
+	}
+	return cfp, nil
+}
```

Comment:
I think, this could be replace by just:
```
if cfp.bbsCache, err := cloudfoundry.GetGlobalBBSCache(), err != nil {
    return nil, err
}
return cfp, nil
```

Reply:
You're right, I'll do that.